pkg/network/prober/prober.go

```diff
@@ -96,7 +106,18 @@ func Do(ctx context.Context, transport http.RoundTripper, target string, ops ...
 			}
 		}
 	}
-	return resp.StatusCode == http.StatusOK, nil
+	if resp.StatusCode == http.StatusOK {
+		return true, nil
+	}
+	for _, op := range ops {
+		if vo, ok := op.(Exception); ok {
+			ok, err := vo(resp, body)
```

Comment:
This seems to return success on the first check. Probably should follow the same path as others.

Reply:
See, comment above, implement this as a Verifier then this code is not needed.